commit 9f6a5eadcd9c6625f78ce676bda723763923f7fa (refs/changes/94/2594/1)
Author: Andrey Yatskov <andrey.yatskov@silvertreesystems.com>
Date:   2017-09-18T11:51:18+03:00 (2 years, 1 month ago)
    
    ADMINUI-2367 AdminUI has trouble talking to UFDS master after network partition

diff --git a/lib/ufds.js b/lib/ufds.js
index 9c20ba4b..62f2a754 100644
--- a/lib/ufds.js
+++ b/lib/ufds.js
@@ -39,10 +39,15 @@ module.exports = {
 
         ufds.on('timeout', function (msg) {
             log.warn({message: msg}, 'UFDS client timeout (recycling)');
-            ufds.close();
         });
 
+        var connectTimout = setTimeout(function () {
+            log.error('Connecting to UFDS failed ', config.url);
+            reconnect();
+        }, config.connectTimeout);
+
         ufds.once('connect', function () {
+            clearTimeout(connectTimout);
             log.info('Connected to UFDS: ', config.url);
             ufds.removeAllListeners('error');
 
